tests/derive/default_value.rs

```diff
@@ -44,6 +46,124 @@ fn auto_default_value_t() {
     assert!(help.contains("[default: 0]"));
 }
 
+#[test]
+fn default_values_t() {
+    #[derive(Parser, PartialEq, Debug)]
+    struct Opt {
+        #[clap(default_values_t = vec![1, 2, 3])]
+        arg1: Vec<i32>,
+
+        #[clap(long, default_values_t = &[4, 5, 6])]
+        arg2: Vec<i32>,
+
+        #[clap(long, default_values_t = [7, 8, 9])]
+        arg3: Vec<i32>,
+
+        #[clap(long, default_values_t = 10..=12)]
+        arg4: Vec<i32>,
+
+        #[clap(long, default_values_t = vec!["hello".to_string(), "world".to_string()])]
+        arg5: Vec<String>,
+
+        #[clap(long, default_values_t = &vec!["foo".to_string(), "bar".to_string()])]
+        arg6: Vec<String>,
+    }
+    assert_eq!(
+        Opt {
+            arg1: vec![1, 2, 3],
+            arg2: vec![4, 5, 6],
+            arg3: vec![7, 8, 9],
+            arg4: vec![10, 11, 12],
+            arg5: vec!["hello".to_string(), "world".to_string()],
+            arg6: vec!["foo".to_string(), "bar".to_string()],
+        },
+        Opt::try_parse_from(&["test"]).unwrap()
+    );
+    assert_eq!(
+        Opt {
+            arg1: vec![1],
+            arg2: vec![4, 5, 6],
+            arg3: vec![7, 8, 9],
+            arg4: vec![10, 11, 12],
+            arg5: vec!["hello".to_string(), "world".to_string()],
+            arg6: vec!["foo".to_string(), "bar".to_string()],
+        },
+        Opt::try_parse_from(&["test", "1"]).unwrap()
+    );
+    assert_eq!(
+        Opt {
+            arg1: vec![1, 2, 3],
+            arg2: vec![4, 5, 6],
+            arg3: vec![7, 8, 9],
+            arg4: vec![42, 15],
+            arg5: vec!["baz".to_string()],
+            arg6: vec!["foo".to_string(), "bar".to_string()],
+        },
+        Opt::try_parse_from(&["test", "--arg4", "42", "--arg4", "15", "--arg5", "baz"]).unwrap()
+    );
+
+    let help = utils::get_long_help::<Opt>();
+    assert!(help.contains("[default: 1 2 3]"));
+}
+
+#[test]
+fn default_value_os_t() {
```

Comment:
looks like https://github.com/clap-rs/clap/commit/01a3ea425fd8ab17997f7ee0e55fd24ccf9f641c deleted this test? not sure if that was intended so added it back here

Reply:
Yeah, I'm a bit confused as to why I removed that test.